google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -143,17 +145,18 @@ public GapicSpannerRpc(SpannerOptions options) throws IOException {
             mergedHeaderProvider.getHeaders(),
             internalHeaderProviderBuilder.getResourceHeaderKey());
 
-    // TODO(pongad): make channel pool work
-
     // TODO(pongad): make RPC logging work (formerly LoggingInterceptor)
     // TODO(pongad): add watchdog
     // TODO(pongad): make error augmentation work (formerly SpannerErrorInterceptor)
 
     TransportChannelProvider channelProvider =
```

Comment:
We also want the user to be able to plug in there own channel provider if needed. Is that possible?

Reply:
It would be easy to make `channelProvider` configurable. I'll add a TODO here and work on it in a later PR.